### PR TITLE
Bgreen/suggested

### PR DESCRIPTION
**Pull request by actaeon**
@didip 

One bug fix and one cosmetic? change for docker\* readers.

### CODE REVIEW

File: readers/docker/containers_cpu_linux.go [hunk 1]
```diff
@@ -55,7 +55,7 @@ func (m *DockerContainersCpu) Run() error {
 
 			data, err := gopsutil_docker.CgroupCPU(containerDir, m.CgroupBasePath)
 			if err == nil {
-				m.Data[container.Image+"-"+container.ID] = data
+				m.Data[container.Names[0]] = data
```
Comment (didip) [inline readers/docker/containers_cpu_linux.go:58]: What if Names is of length 0? Should we handle that case? E.g. `if err == nil && len(container.Names) > 0`?

File: readers/docker/containers_memory_linux.go [hunk 2]
```diff
@@ -52,7 +52,7 @@ func (m *DockerContainersMemory) Run() error {
 
 			data, err := gopsutil_docker.CgroupMem(containerDir, m.CgroupBasePath)
 			if err == nil {
-				m.Data[container.Image+"-"+container.ID] = data
+				m.Data[container.Names[0]] = data
```
Comment (didip) [inline readers/docker/containers_memory_linux.go:55]: Same here, what if Names is of length 0? Should we handle that case?

File: readers/docker/containers.go [hunk 3]
```diff
@@ -31,7 +31,7 @@ func (dc *DockerContainers) Run() error {
 
 	for _, container := range containers {
 		if container.ID != "" && container.NiceImageName != "" {
-			dc.Data[container.NiceImageName+"-"+container.ID] = container
+			dc.Data[container.Name] = container
```
Comment (didip) [inline readers/docker/containers.go:34]: Are you sure that container.Name returns the nice tag name? Last time I pulled container.Name, it returned SHA.

That said, `&& container.NiceImageName != ""` needs to be changed to `&& container.Name != ""`

File: readers/docker/containers_cpu_linux.go (hunk 1 above)
Comment (actaeon) [inline readers/docker/containers_cpu_linux.go:58]: Ok, I'll default to the existing method in that case, does this sound ok?

File: readers/docker/containers.go (hunk 3 above)
Comment (actaeon) [inline readers/docker/containers.go:34]: I'll double check .Name

Bah, I'll fix this other error too.

File: readers/docker/containers_cpu_linux.go (hunk 1 above)
Comment (didip) [inline readers/docker/containers_cpu_linux.go:58]: kk

File: libdocker/libdocker.go [hunk 4]
```diff
@@ -185,9 +186,8 @@ func AllInspectedContainers(endpoint string) ([]*CompleteDockerContainer, error)
 		go func(container *CompleteDockerContainer) {
 			defer wg.Done()
 
-			fullDescContainer, err := client.InspectContainer(shortDescContainer.ID)
+			fullDescContainer, err := client.InspectContainer(container.ID)
```
Comment (didip) [inline libdocker/libdocker.go:189]: I am not sure if `shortDescContainer.ID == fullDescContainer.ID`. If they are the same then this is OK.

File: readers/docker/containers_cpu_linux.go [hunk 5]
```diff
@@ -55,7 +55,7 @@ func (m *DockerContainersCpu) Run() error {
 
 			data, err := gopsutil_docker.CgroupCPU(containerDir, m.CgroupBasePath)
 			if err == nil {
-				m.Data[container.Image+"-"+container.ID] = data
```
Comment (didip) [inline readers/docker/containers_cpu_linux.go:58]: One more thing, the reason why the key is named this way is to ensure uniqueness. If we are changing to only use image name, then we need to include tag as well.

Comment (actaeon) [inline readers/docker/containers_cpu_linux.go:58]: What I'm using is the name of the container, not the image. The docker docs say that the container name is guaranteed to be unique. I'm pretty sure it's also required. This makes it an ideal hash key, no?

File: libdocker/libdocker.go (hunk 4 above)
Comment (actaeon) [inline libdocker/libdocker.go:189]: shortDescContainer.ID comes from client.ListContainer(), fullDescContainer.ID comes from client.InspectContainer(), both represent the ID of the container being listed / inspected. I think this is ok

File: readers/docker/containers.go (hunk 3 above)
Comment (actaeon) [inline readers/docker/containers.go:34]: From a docker inspect:

```
[root@chi-staging-docker-10 ~]# docker inspect 1a5c607fe2e9
[{

(...)

    "Id": "1a5c607fe2e9190b90cd56238ba08e0c4b642d69c8881643e67fa90738d40566",
    "Image": "b4e2252cc3d49667900cecf928c0103a60eb17f60cfbb0094c28cd7a822f41d8",
    "MountLabel": "",
    "Name": "/db-status-baa0d5c74d1621",

(...)

}]
```

The Name key is the container name.

**Comment by actaeon** [comment]
@didip, I fixed the issues that you pointed out, and double-checked the items you had questions for.

**Pull request closed without merging**